components/brave_rewards/resources/extension/brave_rewards/utils.ts

```diff
@@ -58,3 +58,37 @@ export const getGrant = (grant?: RewardsExtension.GrantInfo) => {
 
   return grant
 }
+
+export const setBadgeText = (state: RewardsExtension.State, verified: boolean = false, tabId: number = -1) => {
+  let text = ''
+
+  if (state && state.notifications && !verified) {
+    const count = Object.keys(state.notifications).length
+    if (count > 0) {
+      text = count.toString()
+    }
+  }
+
+  let data: chrome.browserAction.BadgeTextDetails = {
+    text
+  }
+
+  if (tabId !== -1) {
+    data.tabId = tabId
+    chrome.browserAction.setBadgeBackgroundColor({
+      color: verified ? '#4C54D2' : '#FB542B',
+      tabId
+    })
```

Comment:
Curious: If this ternary sets the badge color to orange in all instances except verified, do we need the `else` section of this? Or is it a failsafe?

Reply:
The else is if we are settings the badge text / icon default, and not overriding for a specific tab (to put the verified badge text)